src/Acme/boletinesBundle/Controller/InstitucionController.php

```diff
@@ -96,16 +101,24 @@ public function editAction($id = null, Request $request = null)
 
     private function editEntity($data, $id)
     {
-        $em = $this->getDoctrine()->getManager();
-        $institucion = $em->getRepository('BoletinesBundle:Institucion')->findOneBy(array('idInstitucion' => $id));
+        try{
+            $em = $this->getDoctrine()->getManager();
 
-        $institucion->setNombreInstitucion($data->request->get('nombreInstitucion'));
-        $institucion->setDireccionInstitucion($data->request->get('direccionInstitucion'));
-        $institucion->setEmailInstitucion($data->request->get('emailInstitucion'));
-        $institucion->setTelefonoInstitucion($data->request->get('telefonoInstitucion'));
+            $institucion = $em->getRepository('BoletinesBundle:Institucion')->findOneBy(array('idInstitucion' => $id));
 
-        $em->persist($institucion);
-        $em->flush();
+            $institucion->setNombreInstitucion($data->request->get('nombreInstitucion'));
+            $institucion->setDireccionInstitucion($data->request->get('direccionInstitucion'));
+            $institucion->setEmailInstitucion($data->request->get('emailInstitucion'));
+            $institucion->setTelefonoInstitucion($data->request->get('telefonoInstitucion'));
+
+            $em->persist($institucion);
+            $em->flush();
+
+        } catch (\Exception $e) {
+            $error = $e->getMessage();
+            // si quieren ver el error hacen un print de esto y exit : print_r( $error ); exit();
+            return false;
```

Comment:
Seria mejor creo q dispare un render a una pagina de error html 503, no?

---- Mensaje original ---

De: "kstoriadis" <notifications@github.com>
Enviado: 19 de mayo de 2015 9:05 AM
Para: "nlattessi/boletin-electronico" <boletin-electronico@noreply.github.com>
Asunto: Re: [boletin-electronico] Try catch example (#17)

>
> -        $em->persist($institucion);
> -        $em->flush();
> +            $institucion->setNombreInstitucion($data->request->get('nombreInstitucion'));
> +            $institucion->setDireccionInstitucion($data->request->get('direccionInstitucion'));
> +            $institucion->setEmailInstitucion($data->request->get('emailInstitucion'));
> +            $institucion->setTelefonoInstitucion($data->request->get('telefonoInstitucion'));
> +
> +            $em->persist($institucion);
> +            $em->flush();
> +
> +        } catch (\Exception $e) {
> +            $error = $e->getMessage();
> +            // si quieren ver el error hacen un print de esto y exit : print_r( $error ); exit();
> +            return false;

Devolver false cuando espera un objeto institución no lo va a hacer volar en mil pedazos?

---
Reply to this email directly or view it on GitHub:
https://github.com/nlattessi/boletin-electronico/pull/17/files#r30592255